FaunaDB.Client/Client/DefaultClientIO.cs

```diff
@@ -34,11 +36,14 @@ public DefaultClientIO(Uri domain, TimeSpan timeout, string secret)
             client.DefaultRequestHeaders.AcceptEncoding.Add(new StringWithQualityHeaderValue("gzip"));
             client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
 
-            authHeader = AuthString(secret);
+            authHeader = AuthHeader(secret);
         }
 
         public IClientIO NewSessionClient(string secret) =>
-            new DefaultClientIO(client, secret);
+            new DefaultClientIO(client, AuthHeader(secret), useRecursiveRefs);
+
+        public IClientIO NewWithRecursiveRefs(bool useRecursive) =>
```

Comment:
I think it is unlikely that people will bounce back and forth different versions of the db api during the application execution. That could also be risky in terms of data integrity, for example, if we require a data migration from one version to another.

Having methods that can induce the user to turn on and off a api version is a bad patter, IMO.